src/android/CodePush.java

```diff
@@ -124,32 +137,36 @@ private boolean execInstall(CordovaArgs args, CallbackContext callbackContext) {
         return true;
     }
 
-    private void markUpdateAndStartTimer(int updateSuccessTimeoutInMillis) {
-    /* this flag will clear when reloading the plugin */
-        this.didUpdate = true;
-
-        if (updateSuccessTimeoutInMillis > 0) {
-            startRollbackTimeout(updateSuccessTimeoutInMillis);
-        } else {
-            cleanOldPackageSilently();
+    private boolean execRestartApplication(CordovaArgs args, CallbackContext callbackContext) {
+        try {
+            /* check if we have a deployed package already */
+            CodePushPackageMetadata deployedPackageMetadata = this.codePushPackageManager.getCurrentPackageMetadata();
+            if (deployedPackageMetadata != null) {
+                callbackContext.success();
+                didStartApp = false;
+                onStart();
+            } else {
+                callbackContext.error("The application has no installed updates.");
```

Comment:
I changed the implementation to always restart. I will push the changes in a bit.